### PR TITLE
Update to GoogleTest 1.11.0

### PR DESCRIPTION
**Pull request by yoni206**
When trying to build unit tests on m1 mac an error about deprecated methods fails the compilation of gtest.
This PR suppresses these checks on a mac, which makes the installation of gtest work smoothly.

### CODE REVIEW

File: cmake/FindGTest.cmake [hunk 1]
```diff
@@ -43,6 +48,7 @@ if(NOT GTest_FOUND_SYSTEM)
         CMAKE_ARGS
           -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
           -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}
+	  -DCMAKE_CXX_FLAGS=${CARGS}
```
Comment (nafur) [inline cmake/FindGTest.cmake:51]:
```suggestion
	      -DCMAKE_CXX_FLAGS=${CARGS}
```

**Comment by mpreiner** [comment]
@yoni206 What's the error message you get when trying to compile googletest?

**Comment by yoni206** [comment]
@mpreiner 
It was several `error: definition of implicit copy constructor` errors. Full log here:

[tmp.txt](https://github.com/cvc5/cvc5/files/8774135/tmp.txt)

**Comment by nafur** [comment]
I wonder whether we should generally suppress the warnings, or make it more specific (to only suppress on M1, not for macOS in general)

**Comment by yoni206** [comment]
I suggest only on M!. pushed.

File: cmake/FindGTest.cmake [hunk 2]
```diff
@@ -34,6 +34,11 @@ if(NOT GTest_FOUND_SYSTEM)
 
     set(GTest_VERSION "1.10.0")
 
+    # supress all warning on m1 mac
+    if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
+      set(CARGS -w)
```
Comment (4tXJ7f) [inline cmake/FindGTest.cmake:39]: Let's only disable the specific warning that causes troubles and add a comment about why we are disabling that warning (if there is an issue on their issue tracker, we should link to it).

**Review by 4tXJ7f — CHANGES_REQUESTED** [review]

**Comment by yoni206** [comment]
@mpreiner thanks. That's a much better solution, and it works. [ad5ee52](https://github.com/cvc5/cvc5/pull/8813/commits/ad5ee5250f65c7c565c4de3a695e0af64be6fb18)

**Review by 4tXJ7f — APPROVED** [review]
LGTM

**Pull request merged**